### PR TITLE
[16.01] Full merge of 15.10 into 16.01.

### PR DESCRIPTION
**Pull request by jmchilton**

### CODE REVIEW

File: lib/galaxy/tools/__init__.py [hunk 1]
```diff
@@ -1687,6 +1687,7 @@ def check_and_update_param_values_helper( self, inputs, values, trans, messages,
                             messages[ input.name ].append( rep_dict )
                             self.check_and_update_param_values_helper( input.inputs, {}, trans, rep_dict, context, rep_prefix, allow_workflow_parameters=allow_workflow_parameters )
                 elif isinstance( input, Section ):
+                    messages[ input.name ] = {}
```
Comment (nsoranzo) [inline lib/galaxy/tools/__init__.py:1690]: This line and 1720 should probably be double-checked by @blankenberg and @guerler which modified them a few times between the 3rd and 14th of December.

**Comment by martenson** [comment]
we should merge this to prevent future conflicts

seems ok to me but others please review
ping @galaxyproject/core

**Comment by jmchilton** [comment]
Already conflicted :smile:.

**Comment by dannon** [comment]
@jmchilton Yep, that was me.  I'll review and merge this manually to resolve the new conflicts I made.

**Comment by jmchilton** [comment]
Pushed a merged - if you'd like to use the button instead.

**Comment by martenson** [comment]
I clicked da button

**Comment by dannon** [comment]
@martenson Thought you were looking for another review, but cool.

**Comment by martenson** [comment]
@dannon I think we are always looking for reviews. Post-mortem(merge) works too though. In this case it prevented future conflicts and additional work.

**Comment by dannon** [comment]
Sure, I was just trying to help out and review like you requested (seemed like you were asking for more review prior to merge, explicitly) and said I'd handle the conflict locally.  Not a big deal, just not a huge fan of the "Sure, I'm reviewing and resolving this" ... "I clicked da button" interaction there.

**Comment by martenson** [comment]
There were no conflicts when I merged. The 'I clicked da button' was meant
as a notification for you that you do not need to resolve locally.

**Pull request merged**